swupd.zsh

```diff
@@ -0,0 +1,281 @@
+#compdef swupd
+# -----------------------------------------------------------------------
+#   Software Updater - autocompletion script
+#
+#   Author: Lucius Hu - http://github.com/lebensterben
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, version 2 or later of the License.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------
+
+local context curcontext="$curcontext" state state_descr line subcmd ops ret=1
+local -A opt_args
+local -A val_args
+local installed # cache for installed bundles
+
+(( $+functions[_swupd_bundle_add] )) ||
+  _swupd_bundle_add () {
+    local bundles
+    local MoM
+    if [ -r /var/tmp/swupd/Manifest.MoM ]; then
+      MoM=/var/tmp/swupd/Manifest.MoM
+    elif [ -r /var/lib/swupd/version ] && [ -r /var/lib/swupd/"$(cat /var/lib/swupd/version)"/Manifest.MoM ]; then
+      MoM=/var/lib/swupd/"$(cat /var/lib/swupd/version)"/Manifest.MoM
```

Comment:
This file is only accessible as root. I didn't test this, isn't this going to print errors when running as a regular user?

Reply:
The `[ -r foo ]` test reruns true if the file is readable by current user, from which a list of bundles are added as candidates for completion.

The bottom line is, even if none of those are readble, it shall not report error while users will just not seeing any completion candidates here.

Reply:
You're right about that. But there's another problem.

It looks like [ -n $"MoM" ] is true when running as a regular user in the case /var/tmp/swupd doesn't exist. And I'm seeing the error: "sed: can't read : No such file or directory"

Reply:
I fixed the typo. It should be `[ -n "$MoM" ]`

Reply:
Nice. Works for me now

Reply:
If there is any change to `swupd` command, please let me know and I will update this `zsh` completion.